hail/src/test/scala/is/hail/annotations/UnsafeSuite.scala

```diff
@@ -349,7 +349,55 @@ class UnsafeSuite extends SparkSuite {
     }
     p.check()
   }
-  
+
+  @Test def testRegionsAllocation() {
+    val pool = RegionPool.get
+
+    case class Counts(regions: Int, freeRegions: Int, freeBlocks: Int) {
+      def allocateRegion(): Counts =
+        if (freeRegions == 0)
+          copy(regions = regions + 1)
+        else copy(freeRegions = freeRegions - 1)
+
+      def refreshRegion(nDependentRegions: Int = 0, nBlocks: Int = 0): Counts =
+        if (freeRegions == 0)
+          Counts(regions + 1, freeRegions + nDependentRegions + 1, freeBlocks - nBlocks)
```

Comment:
`freeRegions` is zero so seems odd to include it in the sum here.

Reply:
If I have no free regions, why do I gain one free region? Aren't I deleting the current one and replacing it with a fresh one, ergo no net gain?

Reply:
I'm creating the region first (since I need to know what pool it belongs to), and then freeing the old region. I think I can probably pull out the pool it belongs to, delete the region, and then ask for a fresh one, in which case it will behave as you describe.